server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -237,8 +241,16 @@ public abstract class BlobStoreRepository extends AbstractLifecycleComponent imp
      */
     public static final Setting<Boolean> COMPRESS_SETTING = Setting.boolSetting("compress", true, Setting.Property.NodeScope);
 
+    /**
+     * Setting to disable caching of the latest repository data.
+     */
+    public static final Setting<Boolean> CACHE_REPOSITORY_DATA =
+        Setting.boolSetting("cache_repository_data", true, Setting.Property.Deprecated);
```

Comment:
For ES, until now, this setting is marked as deprecated, should we also keep it like that?

Reply:
I'd tend to remove the setting altogether and always cache. Or what are the reasons why someone would want to disable this?

Reply:
Actually let's keep it for now to reduce the chance of conflicts on other backports. I'll add a note to the 5.1 board that we should revisit this before the release

Reply:
@mfussenegger @matriv As I understand from the original PR of ES, the use-case for adding an undocumented internal only setting is test-cases, see https://github.com/elastic/elasticsearch/pull/52341#discussion_r381887857.
I wonder a bit why we decided to make it settable as an optional repository setting.
Additionally, it seems that no existing test requires disabling this cache so I see no reason to keep such a setting.

Reply:
See my comment just above:

> Actually let's keep it for now to reduce the chance of conflicts on other backports. I'll add a note to the 5.1 board that we should revisit this before the release

Reply:
Ok make sense related to conflicts, sorry for missing this point. But why do we support it as an optional repository setting?

Reply:
Adding it to the `optionalSettings` of each repository is neither related to any possible conflicts as this is our custom code nor needed by any test afaik, such I don't understand the reasoning for adding it. Anyhow, if we remove it completely at a later point, my concern is gone.